internal/config/config.go

```diff
@@ -216,42 +160,17 @@ func parseNodeSelector(ns *nodeSelector) (labels.Selector, error) {
 	return metav1.LabelSelectorAsSelector(sel)
 }
 
-func parseHoldTime(ht string) (time.Duration, error) {
-	if ht == "" {
-		return 90 * time.Second, nil
-	}
-	d, err := time.ParseDuration(ht)
-	if err != nil {
-		return 0, fmt.Errorf("invalid hold time %q: %s", ht, err)
-	}
-	rounded := time.Duration(int(d.Seconds())) * time.Second
+func validateHoldTime(ht time.Duration) error {
+	rounded := time.Duration(int(ht.Seconds())) * time.Second
 	if rounded != 0 && rounded < 3*time.Second {
-		return 0, fmt.Errorf("invalid hold time %q: must be 0 or >=3s", ht)
-	}
-	return rounded, nil
-}
-
-func parseKeepaliveTime(ht time.Duration, ka string) (time.Duration, error) {
-	// If keepalive time not set lets use 1/3 of holdtime.
-	if ka == "" {
-		return ht / 3, nil
-	}
-	d, err := time.ParseDuration(ka)
-	if err != nil {
-		return 0, fmt.Errorf("invalid keepalive time %q: %s", ka, err)
+		return fmt.Errorf("invalid hold time %q: must be 0 or >=3s", ht)
 	}
-	rounded := time.Duration(int(d.Seconds())) * time.Second
-	return rounded, nil
+	return nil
 }
 
 // Parse loads and validates a Config from bs.
-func Parse(bs []byte, validate Validate) (*Config, error) {
-	var raw configFile
-	if err := yaml.UnmarshalStrict(bs, &raw); err != nil {
-		return nil, fmt.Errorf("could not parse config: %s", err)
-	}
-
-	err := validate(&raw)
+func For(crds CRSConfig, validate Validate) (*Config, error) {
```

Comment:
Wondering about the name of the function, why For and not Parse?

Reply:
Because no parsing is happening here, we are getting a Config for what we are passing.
Before we were parsing a yaml file, so the old name made sense.

Reply:
I just think the name is unclear for what is done here. But maybe that's just me.
It is parsing objects. For example: `parsed, err := bfdProfileFromCR(bfd)`

Reply:
It's not parsing. And from the calling sites, it will look config.For(crds) which I think is clear.